server/src/main/java/org/opensearch/extensions/ExtensionRequest.java

```diff
@@ -24,41 +25,53 @@
  */
 public class ExtensionRequest extends TransportRequest {
     private static final Logger logger = LogManager.getLogger(ExtensionRequest.class);
-    private ExtensionsManager.RequestType requestType;
+    private final ExtensionsManager.RequestType requestType;
+    private final Optional<String> uniqueId;
 
     public ExtensionRequest(ExtensionsManager.RequestType requestType) {
+        this(requestType, null);
+    }
+
+    public ExtensionRequest(ExtensionsManager.RequestType requestType, Optional<String> uniqueId) {
         this.requestType = requestType;
+        this.uniqueId = uniqueId;
     }
 
     public ExtensionRequest(StreamInput in) throws IOException {
-        super(in);
```

Comment:
I believe Requests require the `super(in)` while responses do not.  This should definitely match the `writeTo` implementation; I don't expect it will pass tests without it matching.